### PR TITLE
Update Bio WP diagrams

### PR DESCRIPTION
**Pull request by TorkelE**
Updates the WP diagrams for the bio models.

### CODE REVIEW

File: benchmarks/Bio/BCR.jmd [hunk 1]
```diff
@@ -26,6 +26,7 @@ tf       = 100000.0
 # generate ModelingToolkit ODEs
 @timeit to "Parse Network" prnbng = loadrxnetwork(BNGNetwork(), joinpath(datadir, "bcr.net"))
 rn    = prnbng.rn
+@unpack Activated_Syk,Ig_alpha_P,Ig_alpha_PP,Ig_beta_PP,Activated_Lyn,Autoinhibited_Lyn,Activated_Fyn,Autoinhibited_Fyn,PAG1_Csk = rn; obs = [Activated_Syk,Ig_alpha_P,Ig_alpha_PP,Ig_beta_PP,Activated_Lyn,Autoinhibited_Lyn,Activated_Fyn,Autoinhibited_Fyn,PAG1_Csk]
```
Comment (isaacsas) [inline benchmarks/Bio/BCR.jmd:29]:
```suggestion
@unpack Activated_Syk,Ig_alpha_P,Ig_alpha_PP,Ig_beta_PP,Activated_Lyn,Autoinhibited_Lyn,Activated_Fyn,Autoinhibited_Fyn,PAG1_Csk = rn; obs = [Activated_Syk,Ig_alpha_P,Ig_alpha_PP,Ig_beta_PP,Activated_Lyn,Autoinhibited_Lyn,Activated_Fyn,Autoinhibited_Fyn,PAG1_Csk]
obs = observed(rn)
```

Comment (TorkelE) [reply benchmarks/Bio/BCR.jmd:29]: good one

**Comment by TorkelE** [comment]
is there a way to view these? I try https://benchmarks.sciml.ai/previews/PR527/ but the page is not found.

**Comment by ChrisRackauckas** [comment]
Please branch this off of master.

**Comment by ChrisRackauckas** [comment]
Yes just look at the build artifacts

**Comment by TorkelE** [comment]
Sorry, I'm afraid I do not understand. I tried finding build artefacts via Actions (something I found when looking for how to do only), but there there is nothing for PRs. I also tried click "Details" at:
image
and get to:
image
but everything says 0 artifacts, or the top I cannot click.
(but maybe this has to do with the blocking thing, which has to do with the branch of master?)

**Comment by ChrisRackauckas** [comment]
That's because it hasn't ben run yet. I wasn't going to unblock CI because I thought you'd re-bracnh from master.

**Comment by TorkelE** [comment]
> Please branch this off of master.

Sorry, again I do not think I understand. So I created a fork at https://github.com/TorkelE/SciMLBenchmarks.jl, which I then created a new branch locally:
image

Do you mean that I should create this branch from something that is not the master branch? I am unsure what the proper workflow for these things are, tried to read some GitHub resources, but in the end went with what allowed me to create a PR.

**Comment by TorkelE** [comment]
Ahh, you mean that my branch is not from master, and that is what it should be? Will try to fix.

**Comment by TorkelE** [comment]
Ok, I _think_ I get what was going on now (and fixed it). Thanks!

File: benchmarks/Bio/BCR.jmd [hunk 2]
```diff
@@ -90,24 +97,62 @@ test_sol  = TestSolution(sol)
...
+#### Declare pre-conditioners
+```julia
+using IncompleteLU, LinearAlgebra
+
+jaccache = sparsejacprob.f.jac(oprob.u0,oprob.p,0.0)
+W = I - 1.0*jaccache
+prectmp = ilu(W, τ = 50.0)
+preccache = Ref(prectmp)
+
+τ1 = 1e2
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:109]: These two tau choices are not the same?

File: benchmarks/Bio/BCR.jmd [hunk 3]
```diff
@@ -90,24 +97,62 @@ test_sol  = TestSolution(sol)
...
-          Dict(:alg=>KenCarp47(autodiff=false)),
-          #Dict(:alg=>RadauIIA5(autodiff=false)),
-          #Dict(:alg=>lsoda()),
+          Dict(:alg=>lsoda(), :prob_choice => 1),
+          Dict(:alg=>CVODE_BDF(linear_solver=:GMRES,prec=precilu_fceri,psetup=psetupilu_fceri,prec_side=1), :prob_choice => 2),
+          Dict(:alg=>TRBDF2(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2),
+          Dict(:alg=>QNDF(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2),
+          Dict(:alg=>FBDF(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2),
+          Dict(:alg=>KenCarp4(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2),
+          Dict(:alg=>Rosenbrock23(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2)
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:155]: Why Rosenbrock23? Sounds like Rodas5P would be better off here? Tolerances are too low.

Comment (ChrisRackauckas) [reply benchmarks/Bio/BCR.jmd:155]: This is missing any usage of the sparse factorization methods. That seems like a big omission.

File: benchmarks/Bio/egfr_net.jmd [hunk 4]
```diff
@@ -103,6 +109,8 @@ Finally, we generate a work-precision diagram for the selection of solvers.
 ```julia
 wp = WorkPrecisionSet(oprob,abstols,reltols,setups;error_estimate=:l2,
                       saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e5),numruns=100)
+<<<<<<< HEAD
+<<<<<<< master
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/egfr_net.jmd:113]: merge issue.

File: benchmarks/Bio/egfr_net.jmd [hunk 5]
```diff
@@ -130,8 +138,22 @@ setups = [
...
+
+>>>>>>> update_bio_benchmarks
+names = ["lsoda" "CVODE_BDF (GMRES)" "TRBDF2 (GMRES)" "KenCarp4 (GMRES)" "QNDF (GMRES)" "FBDF (GMRES)" "Rosenbrock23 (GMRES)"]
+markers = [:star4 :circle :diamond :rect :dtriangle :star5 :heptagon]
+plot(wp;label=names,markershape=markers,legendfontsize=7,framestyle=:box,xlimit=(1e-8,1e0),ylimit=(1e1,1e4),gridalpha=0.3,gridlinewidth=2.5)
+
+<<<<<<< HEAD
+>>>>>>> Update models without pre_cond
+=======
+>>>>>>> update_bio_benchmarks
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/egfr_net.jmd:153]: merge issues.

File: benchmarks/Bio/fceri_gamma2.jmd [hunk 6]
```diff
@@ -21,15 +21,32 @@ using DiffEqBase, OrdinaryDiffEq, Catalyst, ReactionNetworkImporters,
 
 gr()
 const to = TimerOutput()
-tf       = 0.5
+tf       = 150.0
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/fceri_gamma2.jmd:24]: why?

File: benchmarks/Bio/fceri_gamma2.jmd [hunk 7]
```diff
@@ -21,15 +21,32 @@ using DiffEqBase, OrdinaryDiffEq, Catalyst, ReactionNetworkImporters,
...
+@timeit to_oprob "ODEProb No Jac" oprob = ODEProblem(osys, Float64[], tspan, Float64[])
+oprob_sparse = ODEProblem(osys, Float64[], tspan, Float64[]; sparse=true)
+
+show(to_parse) 
+show(to_osys) 
+show(to_oprob);
+<<<<<<< HEAD
+>>>>>>> Update Multistate model
+=======
+>>>>>>> update_bio_benchmarks
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/fceri_gamma2.jmd:49]: Merge issues

File: benchmarks/Bio/fceri_gamma2.jmd [hunk 8]
```diff
@@ -80,26 +97,84 @@ test_sol  = TestSolution(sol)
...
-          Dict(:alg=>FBDF(linsolve=KrylovJL_GMRES(),autodiff=false)),
-          Dict(:alg=>KenCarp4(linsolve=KrylovJL_GMRES(),autodiff=false)),
-          Dict(:alg=>Rosenbrock23(linsolve=KrylovJL_GMRES(),autodiff=false)),
+          Dict(:alg=>lsoda(), :prob_choice => 1),
+          Dict(:alg=>CVODE_BDF(linear_solver=:GMRES,prec=precilu_fceri,psetup=psetupilu_fceri,prec_side=1), :prob_choice => 2),
+          Dict(:alg=>TRBDF2(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2),
+          Dict(:alg=>QNDF(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2),
+          Dict(:alg=>FBDF(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2),
+          Dict(:alg=>KenCarp4(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2),
+          Dict(:alg=>Rosenbrock23(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true), :prob_choice => 2)
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/fceri_gamma2.jmd:155]: Should benchmark with and without preconditioners

File: benchmarks/Bio/fceri_gamma2.jmd [hunk 9]
```diff
@@ -80,26 +97,84 @@ test_sol  = TestSolution(sol)
...
+=======
+wp = WorkPrecisionSet([oprob,oprob_sparse],abstols,reltols,setups;error_estimate=:l2,
+>>>>>>> update_bio_benchmarks
                       saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e5),numruns=10)
-plot(wp)
+
+names = ["lsoda" "CVODE_BDF (GMRES, iLU)" "TRBDF2 (GMRES, iLU)" "KenCarp4 (GMRES, iLU)" "QNDF (GMRES, iLU)" "FBDF (GMRES, iLU)" "Rosenbrock23 (GMRES, iLU)"]
+markers = [:star4 :circle :diamond :rect :dtriangle :star5 :heptagon]
+plot(wp;label=names,markershape=markers,legendfontsize=7,framestyle=:box,xlimit=(1e-7,1e-3),ylimit=(1e2,1e6),xticks=[1e-7,1e-6,1e-5,1e-4,1e-3],gridalpha=0.3,gridlinewidth=2.5)
+<<<<<<< HEAD
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/fceri_gamma2.jmd:177]: Merge issues

File: benchmarks/Bio/fceri_gamma2.jmd [hunk 10]
```diff
@@ -124,6 +199,8 @@ setups = [
 wp = WorkPrecisionSet(sparsejacprob,abstols,reltols,setups;error_estimate=:l2,
                       saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e5),numruns=200)
 plot(wp)
+=======
+>>>>>>> update_bio_benchmarks
 ```
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/fceri_gamma2.jmd:204]: Merge issues

File: benchmarks/Bio/multisite2.jmd [hunk 11]
```diff
@@ -85,13 +91,13 @@ test_sol  = TestSolution(sol)
...
-          Dict(:alg=>FBDF(linsolve=KrylovJL_GMRES())),
-          Dict(:alg=>KenCarp4(linsolve=KrylovJL_GMRES())),
-          Dict(:alg=>Rosenbrock23(linsolve=KrylovJL_GMRES())),
+          Dict(:alg=>lsoda()), 
+          Dict(:alg=>CVODE_BDF(linear_solver=:GMRES)), 
+          Dict(:alg=>QNDF()), 
+          Dict(:alg=>Tsit5()), 
+          Dict(:alg=>BS5()), 
+          Dict(:alg=>Vern6()), 
+          Dict(:alg=>Vern8())
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/multisite2.jmd:100]: This was the stiff ODE method benchmark, why was this changed?

File: benchmarks/Bio/multisite2.jmd [hunk 12]
```diff
@@ -102,6 +108,8 @@ Finally, we generate a work-precision diagram for the selection of solvers.
 ```julia
 wp = WorkPrecisionSet(oprob,abstols,reltols,setups;error_estimate=:l2,
                       saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e5),numruns=200)
+<<<<<<< HEAD
+<<<<<<< master
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/multisite2.jmd:112]: Merge issues

File: benchmarks/Bio/multisite2.jmd [hunk 13]
```diff
@@ -129,8 +137,21 @@ setups = [
...
+names = ["lsoda" "CVODE_BDF (GMRES)" "QNDF" "Tsit5" "BS5" "Vern6" "Vern8"]
+markers = [:star4 :circle :dtriangle :octagon :star8 :ltriangle :rtriangle]
+plot(wp;label=names,markershape=markers,legendfontsize=7,framestyle=:box,xlimit=(1e-10,1e-2),ylimit=(1e0,1e2),xticks=[1e-10,1e-8,1e-6,1e-4,1e-2],gridalpha=0.3,gridlinewidth=2.5)
+>>>>>>> Update models without pre_cond
+=======
+
+names = ["lsoda" "CVODE_BDF (GMRES)" "QNDF" "Tsit5" "BS5" "Vern6" "Vern8"]
+markers = [:star4 :circle :dtriangle :octagon :star8 :ltriangle :rtriangle]
+plot(wp;label=names,markershape=markers,legendfontsize=7,framestyle=:box,xlimit=(1e-10,1e-2),ylimit=(1e0,1e2),xticks=[1e-10,1e-8,1e-6,1e-4,1e-2],gridalpha=0.3,gridlinewidth=2.5)
+>>>>>>> update_bio_benchmarks
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/multisite2.jmd:151]: Merge issues.

File: benchmarks/Bio/BCR.jmd (hunk 2 above)
Comment (TorkelE) [reply benchmarks/Bio/BCR.jmd:109]: So, for the CVODE pre-conditioner, in https://benchmarks.sciml.ai/dev/StiffODE/Bruss/ there were two tau values, I tried changing both, but only one actually seemed to affect the runtime (the other one I left at its value of 50).

For the tau between the CVODE and Julia pre-conditioners,  I just tried various values, and the tau's that optimised the runtimes were different between the two.

File: benchmarks/Bio/BCR.jmd (hunk 3 above)
Comment (TorkelE) [reply benchmarks/Bio/BCR.jmd:155]: By sparse factorisation method do you mean linsolver? 

Here I took the algorithms we used in the tests (with the best option for each). I am happy to remove/add some if you tell me what to use where.

File: benchmarks/Bio/fceri_gamma2.jmd (hunk 6 above)
Comment (TorkelE) [reply benchmarks/Bio/fceri_gamma2.jmd:24]: In the Zulip discussion we said we'd run fceri_gamma2 to steady state at 150?

File: benchmarks/Bio/fceri_gamma2.jmd (hunk 8 above)
Comment (TorkelE) [reply benchmarks/Bio/fceri_gamma2.jmd:155]: Will add.

File: benchmarks/Bio/multisite2.jmd (hunk 11 above)
Comment (TorkelE) [reply benchmarks/Bio/multisite2.jmd:100]: I thought we said we'd use explicit methods for the two smaller models in Zulip? I  can move multisite2  back to implicit solvers only, but I think the explicit ones perform slightly better.

File: benchmarks/Bio/fceri_gamma2.jmd (hunk 6 above)
Comment (ChrisRackauckas) [reply benchmarks/Bio/fceri_gamma2.jmd:24]: From the plots, it looks like it starts at steady state?

File: benchmarks/Bio/BCR.jmd (hunk 3 above)
Comment (ChrisRackauckas) [reply benchmarks/Bio/BCR.jmd:155]: Yes. The bruss methods are a good template to look at: https://benchmarks.sciml.ai/stable/StiffODE/Bruss/ . On CVODE, show factorization vs GMRES, with and without preconditioners, with different problem setups. Then do that with one of the good Julia methods. Then take what is performing best and pit them against each other. Basically, a dive into the setup of the method, and then comparison between the good ones. That then paints the whole picture.

File: benchmarks/Bio/multisite2.jmd (hunk 11 above)
Comment (ChrisRackauckas) [reply benchmarks/Bio/multisite2.jmd:100]: There was one benchmark on stiff methods, and one benchmark on the non-stiff methods. This PR updates to make both of them on the non-stiff methods, with the same methods. I don't understand why it would need two different plots of almost the the same thing?

File: benchmarks/Bio/fceri_gamma2.jmd (hunk 6 above)
Comment (TorkelE) [reply benchmarks/Bio/fceri_gamma2.jmd:24]: There was a problem with the old plot where it showed all states, not just the observables. I have changed it so that the plots show the observables instead. When the observables only are shown one can see that it continues until and  SS at around t=150.

Comment (ChrisRackauckas) [reply benchmarks/Bio/fceri_gamma2.jmd:24]: Makes more sense

File: benchmarks/Bio/multisite2.jmd (hunk 11 above)
Comment (TorkelE) [reply benchmarks/Bio/multisite2.jmd:100]: Wait, I am probably missing something. So there is the multistate and multisite2 models  (with the second one being larger). For both explicit methods are better, so I  benchmark these (admittedly the same set, but could differentiate a bit). In each case, I also threw in the best stiff solver just to show how one of these performs.

I think it makes sense to benchmark on both multistate and multisite2.  I could include more/different stiff/non-stiff solvers in each case. I could also make one plot with stiff and one with non-stiff solvers in each case.

Comment (ChrisRackauckas) [reply benchmarks/Bio/multisite2.jmd:100]: it's already setup as a plot of stiff then non-stiff, so I am wondering why that was changed.

Comment (TorkelE) [reply benchmarks/Bio/multisite2.jmd:100]: Sorry, I will go through an check this when I get to this to figure out exactly what I did and why.

Comment (TorkelE) [reply benchmarks/Bio/multisite2.jmd:100]: Not sure if this was something that wasn't there in the version I worked on it, and now it got messed up in the merge. Or I just missed it when I thought I'd use the updated solver set. Re-adding explicit solvers for all benchmarks.

File: benchmarks/Bio/BCR.jmd [hunk 14]
```diff
@@ -90,62 +93,193 @@ test_sol  = TestSolution(sol)
...
-          Dict(:alg=>KenCarp47(autodiff=false)),
-          #Dict(:alg=>RadauIIA5(autodiff=false)),
-          #Dict(:alg=>lsoda()),
+          Dict(:alg=>TRBDF2(linsolve=KLUFactorization())),
+          Dict(:alg=>QNDF(linsolve=KLUFactorization())),
+          Dict(:alg=>FBDF(linsolve=KLUFactorization())),
+          Dict(:alg=>KenCarp4(linsolve=KLUFactorization())),
+          Dict(:alg=>Rosenbrock23(linsolve=KLUFactorization())),
+          Dict(:alg=>Rodas4(linsolve=KLUFactorization())),
+          Dict(:alg=>Rodas5P(linsolve=KLUFactorization()))
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:268]: KLU is automatic so that doesn't matter, but it's fine.

**Comment by isaacsas** [comment]
FCERI has undefvar errors on buildkite?

File: benchmarks/Bio/multisite2.jmd [hunk 15]
```diff
@@ -81,56 +86,88 @@ test_sol  = TestSolution(sol)
...
-          Dict(:alg=>CVODE_Adams()),
-          Dict(:alg=>Tsit5()),
-          Dict(:alg=>BS5()),
-          Dict(:alg=>VCABM()),
-          Dict(:alg=>Vern6()),
-          Dict(:alg=>Vern7()),
-          Dict(:alg=>Vern8()),
-          Dict(:alg=>Vern9()),
+          Dict(:alg=>lsoda()), 
+          Dict(:alg=>CVODE_BDF(linear_solver=:GMRES)),
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/multisite2.jmd:148]: Should add an adams here

**Comment by TorkelE** [comment]
Got a:
image
and
image

Does that mean that a time limit was exceeded? Or did we run into some other normal error

**Comment by ChrisRackauckas** [comment]
Yes time limit. Remove the failing cases.

**Comment by TorkelE** [comment]
I have been trying to figure out what cases  failed, but the log only gives me a 
```
<html>
<body>
<!--StartFragment-->

┌ Warning: Interrupted. Larger maxiters is needed. If you are using an integrator for non-stiff ODEs or an automatic switching algorithm (the default), you may want to consider using a method for stiff equations. See the solver pages for more details (e.g. https://diffeq.sciml.ai/stable/solvers/ode_solve/#Stiff-Problems).
--
  | └ @ SciMLBase /cache/julia-buildkite-plugin/depots/5b300254-1738-4989-ae0a-f4d2d937f953/packages/SciMLBase/xWByK/src/integrator_interface.jl:502
  | ┌ Warning: Interrupted. Larger maxiters is needed. If you are using an integrator for non-stiff ODEs or an automatic switching algorithm (the default), you may want to consider using a method for stiff equations. See the solver pages for more details (e.g. https://diffeq.sciml.ai/stable/solvers/ode_solve/#Stiff-Problems).
  | └ @ SciMLBase /cache/julia-buildkite-plugin/depots/5b300254-1738-4989-ae0a-f4d2d937f953/packages/SciMLBase/xWByK/src/integrator_interface.jl:502
  | ┌ Warning: Interrupted. Larger maxiters is needed. If you are using an integrator for non-stiff ODEs or an automatic switching algorithm (the default), you may want to consider using a method for stiff equations. See the solver pages for more details (e.g. https://diffeq.sciml.ai/stable/solvers/ode_solve/#Stiff-Problems).
  | └ @ SciMLBase /cache/julia-buildkite-plugin/depots/5b300254-1738-4989-ae0a-f4d2d937f953/packages/SciMLBase/xWByK/src/integrator_interface.jl:502
  | # Received cancellation signal, interrupting
  | 🚨 Error: The command exited with status -1

<!--EndFragment-->
</body>
</html>
```
and I cannot find clues of which parts takes time. I tried some other tabs but found nothing, am I missing something?

Also, when I run the buildkite, does it only run the updated Bio parts, or will it run the full SciML benchmarks?

**Comment by ChrisRackauckas** [comment]
It only runs the files and folders you update. Folders for where the project/manifest updates, and files where the .md updates. 

> and I cannot find clues of which parts takes time. I tried some other tabs but found nothing, am I missing something?

Do `show_names = true`

File: benchmarks/Bio/BCR.jmd [hunk 16]
```diff
@@ -74,78 +80,201 @@ show(to)
...
-          Dict(:alg=>KenCarp4(autodiff=false)),
-          Dict(:alg=>KenCarp47(autodiff=false)),
-          #Dict(:alg=>RadauIIA5(autodiff=false)),
-          #Dict(:alg=>lsoda()),
+          Dict(:alg=>lsoda(), :prob_choice => 1),
+          Dict(:alg=>CVODE_BDF(), :prob_choice => 1),
+          Dict(:alg=>CVODE_BDF(linear_solver=:LapackDense), :prob_choice => 1),
+          Dict(:alg=>CVODE_BDF(linear_solver=:GMRES), :prob_choice => 1),
+          Dict(:alg=>CVODE_BDF(linear_solver=:GMRES,prec=precilu,psetup=psetupilu,prec_side=1), :prob_choice => 2),
+          Dict(:alg=>CVODE_BDF(linear_solver=:KLU), :prob_choice => 3),
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:162]: I don't understand the return to autodiff here. Won't think take forever?

Comment (TorkelE) [reply benchmarks/Bio/BCR.jmd:162]: No, you are right about this. I tried to comment out the BCR, but it is probably some autodiff on the fceri that is causing the troubles. I'm home to celebrate my mother's birthday, will update things on Tuesday latest.

Comment (ChrisRackauckas) [reply benchmarks/Bio/BCR.jmd:162]: no worries.

**Comment by ChrisRackauckas** [comment]
What's the current status?

**Comment by TorkelE** [comment]
Out things on hold a bit as I was relocating to Boston. Just arrived now and hope to get it sorted out (in tandem with other practical details of the move).

Given that I'm here now I was also hoping to chat with some people more experienced with build kite for some advice. Unless I get bogged down in finding accommodation, I hope we can finish the catalyst paper asap, so that I can move on fully to other stuff.

File: benchmarks/Bio/BCR.jmd [hunk 17]
```diff
@@ -14,138 +14,322 @@ Catalyst network model to ODEs.
 
 
 ```julia
-using DiffEqBase, OrdinaryDiffEq, Catalyst, ReactionNetworkImporters,
-      Sundials, Plots, DiffEqDevTools, ODEInterface, ODEInterfaceDiffEq,
-      LSODA, TimerOutputs, LinearAlgebra, ModelingToolkit, BenchmarkTools
+if false
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:17]: what? That disables it?

File: benchmarks/Bio/BCR.jmd [hunk 18]
```diff
@@ -14,138 +14,322 @@ Catalyst network model to ODEs.
...
+    setups = [
+            Dict(:alg=>TRBDF2(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+            Dict(:alg=>QNDF(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+            Dict(:alg=>FBDF(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+            Dict(:alg=>KenCarp4(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+            Dict(:alg=>Rosenbrock23(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+            Dict(:alg=>Rodas4(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+            Dict(:alg=>Rodas5P(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true))
+            ];
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:286]: Rodas and Rosenbrock don't make sense on this problem? What is this doing here?

File: benchmarks/Bio/fceri_gamma2.jmd [hunk 19]
```diff
@@ -74,59 +87,275 @@ points.
...
+    ylimit = 10 .^ (floor(log10(minimum(times))), ceil(log10(maximum(times))))
+    return xlimit,ylimit
+end
+```
+
+#### Declare pre-conditioners
+```julia
+if false
+    using IncompleteLU, LinearAlgebra
+
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/fceri_gamma2.jmd:122]: this won't run

File: benchmarks/Bio/BCR.jmd [hunk 20]
```diff
@@ -14,138 +14,322 @@ Catalyst network model to ODEs.
...
+    setups = [
+            Dict(:alg=>TRBDF2(autodiff=false)),
+            Dict(:alg=>QNDF(autodiff=false)),
+            Dict(:alg=>FBDF(autodiff=false)),
+            Dict(:alg=>KenCarp4(autodiff=false)),
+            Dict(:alg=>Rosenbrock23(autodiff=false)),
+            Dict(:alg=>Rodas4(autodiff=false)),
+            Dict(:alg=>Rodas5P(autodiff=false)),
+            Dict(:alg=>ROCK2(autodiff=false)), 
+            Dict(:alg=>ROCK4(autodiff=false))
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:227]: ROCK methods will never finish here. I don't understand what you're doing. This seems copy pasted from other benchmarks. Different benchmark problems are very different!

File: benchmarks/Bio/BCR.jmd (hunk 17 above)
Comment (TorkelE) [reply benchmarks/Bio/BCR.jmd:17]: Basically, tying to figure out what went wrong, I disabled all code in the BCR and fceri models, trying to work myself down to figuring out what the problem is. The remaining models all run relatively fast, so shouldn't be a run time issue.

File: benchmarks/Bio/BCR.jmd (hunk 18 above)
Comment (TorkelE) [reply benchmarks/Bio/BCR.jmd:286]: You are probably right, I'm not terribly familiar with the solvers, sop added Rodas (and Rosenbrock) to all the implicit cases. The idea was to then take the slow solvers and put them in a separate plot. However, I am happy to remove them entirely.

File: benchmarks/Bio/BCR.jmd (hunk 20 above)
Comment (TorkelE) [reply benchmarks/Bio/BCR.jmd:227]: Yes, I think I misunderstood earlier, wasn't sure where ROCK belong so did it for all in the initial run.

**Comment by TorkelE** [comment]
Thanks, this is very useful.

Currently, BCR and fceri are commented out (the `if false` blocks). The remaining models I can run fine on my computer, and if I understand it right, they also seem to complete in buildkite. However, something seems to happen at the end of the log, not sure if it has to do with the weave thing. I tried to look up the error code, but didn't find anything useful, so I tried to raise the issue with buildkite support. Hopefully they say something that can help figure out what is going on.

**Comment by ChrisRackauckas** [comment]
Buildkite support won't do anything here.

@staticfloat can you help us figure out what's going on with the artifacts?

**Comment by staticfloat** [comment]
Can you rebase this on top of the latest `master`?  I think there are important CI fixes that were [merged by Anant about 3 weeks ago](https://github.com/SciML/SciMLBenchmarks.jl/commit/68ec69ca8b7413bd65720dd8276e457c2b5ef509) that are not included in your branch.

**Comment by ChrisRackauckas** [comment]
This PR was in the stone age, so I rebased it.

**Comment by ChrisRackauckas** [comment]
Seemed to build fine

**Comment by TorkelE** [comment]
Yes, seems good now. Have to re-run the small models, and then I will add fceri and then BCR. 

For the small models, I will split of the methods that appear distinctly worse. For BCR and fceri (the two big models) I will remove the Rosenbrock, Rodas, and ROCK solvers.

File: benchmarks/Bio/BCR.jmd [hunk 21]
```diff
@@ -74,78 +81,200 @@ show(to)
...
-Finally we test using the generated sparse analytic Jacobian function.
+#### Declare solvers (using GMRES linear solver, with pre-conditioner).
+
 ```julia
 setups = [
-          #Dict(:alg=>Rosenbrock23(autodiff=false)),
-          Dict(:alg=>TRBDF2(autodiff=false)),
-          Dict(:alg=>QNDF(autodiff=false)),
-          Dict(:alg=>FBDF(autodiff=false)),
-          #Dict(:alg=>CVODE_BDF(linear_solver=:KLU)), # Fails!
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:136]: It was noted already that this solver fails. It shouldn't be in a WPD

File: benchmarks/Bio/BCR.jmd [hunk 22]
```diff
@@ -74,78 +81,200 @@ show(to)
...
+#### Declare solvers.
+
+```julia
+setups = [
+        Dict(:alg=>lsoda(), :prob_choice => 1),
+        Dict(:alg=>CVODE_BDF(), :prob_choice => 1),
+        Dict(:alg=>CVODE_BDF(linear_solver=:LapackDense), :prob_choice => 1),
+        Dict(:alg=>CVODE_BDF(linear_solver=:GMRES), :prob_choice => 1),
+        Dict(:alg=>CVODE_BDF(linear_solver=:GMRES,prec=precilu,psetup=psetupilu,prec_side=1), :prob_choice => 2),
+        Dict(:alg=>CVODE_BDF(linear_solver=:KLU), :prob_choice => 3),
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:171]: So I don't understand why it was added here?

Comment (ChrisRackauckas) [reply benchmarks/Bio/BCR.jmd:171]: I presume that was this failure? https://buildkite.com/julialang/scimlbenchmarks-dot-jl/builds/1142#018473ac-e339-42e6-9a10-5d2cfd70d14f/333-375

File: benchmarks/Bio/BCR.jmd [hunk 23]
```diff
@@ -74,78 +81,200 @@ show(to)
...
+
+
+#### Plot Work-Precision Diagram (using sparse jacobian)
+
+Finally, we generate a work-precision diagram for the selection of solvers.
+```julia
 wp = WorkPrecisionSet(sparsejacprob,abstols,reltols,setups;error_estimate=:l2,
-                      saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e5),numruns=1)
-plot(wp)
+                    saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e12),dtmin=1e-18,numruns=2)
```
Comment (ChrisRackauckas) [inline benchmarks/Bio/BCR.jmd:273]: why was this max iters increased so high? That shows that one of the solvers is probably actually failing and should be removed from the set.

**Comment by ChrisRackauckas** [comment]
Wait what crashed? I don't see a crash in the logs.

**Comment by TorkelE** [comment]
Ahh, didn't know how you set the timeout until I saw your latest PR. Saw that buildkite stopped and presumed it was due to a crash.
